Fix initial build of ghc-pwd due to missing libiconv

ghc-pwd fails to build because it is built using ghc-bootstrap, which ships
libHSbase-$version.a, which references symbols from libiconv (and actually the
system libiconv, because the MacPorts use uses _libiconv as symbol names, but
the archive uses _iconv). Fix this by explicitly linking against the system
libiconv for this initial build of ghc-pwd.

Upstream-Status: Inappropriate (/usr/lib/libiconv.dylib is system-specific; a generic solution would be needed)
--- configure.orig	2016-05-20 23:40:29.000000000 +0200
+++ configure	2016-05-20 23:41:06.000000000 +0200
@@ -4149,7 +4149,7 @@
 $as_echo "$as_me: Building in-tree ghc-pwd" >&6;}
                 rm -rf utils/ghc-pwd/dist-boot
     mkdir  utils/ghc-pwd/dist-boot
-                GHC_LDFLAGS=`perl -e 'foreach (@ARGV) { print "-optl$_ " }' -- $LDFLAGS`
+                GHC_LDFLAGS=`perl -e 'foreach (@ARGV) { print "-optl$_ " }' -- $LDFLAGS /usr/lib/libiconv.dylib`
         if ! "$WithGhc" $GHC_LDFLAGS -v0 -no-user-$GHC_PACKAGE_DB_FLAG -hidir utils/ghc-pwd/dist-boot -odir utils/ghc-pwd/dist-boot -stubdir utils/ghc-pwd/dist-boot --make utils/ghc-pwd/Main.hs -o utils/ghc-pwd/dist-boot/ghc-pwd
     then
         as_fn_error $? "Building ghc-pwd failed" "$LINENO" 5
